### PR TITLE
New Stack Editor: Save Flow

### PR DESCRIPTION
**Pull request by gokmen**
### Save Flow

ref: http://recordit.co/4S0eusMbGE

### Credential Warnings on Toolbar Banner
http://recordit.co/AzDNq50kEZ

### CODE REVIEW

File: client/app/lib/util/stacks/updatecustomvariable.coffee [hunk 1]
```diff
@@ -59,8 +59,7 @@ module.exports = updateCustomVariable = (options, callback) ->
         createAndUpdate { provider, title, meta, stackTemplate }, callback
       else
         credential.update { meta, title }, (err) ->
-          shareWithGroup credential, ->
-            callback err, stackTemplate
+          shareWithGroup credential, callback
```
Comment (usirin) [inline client/app/lib/util/stacks/updatecustomvariable.coffee]: Do we really want to share it with group, if there is an error?

File: client/new-stack-editor/lib/controllers/logs.coffee [hunk 2]
```diff
@@ -21,7 +23,13 @@ module.exports = class LogsController extends BaseController
 
     content = stringify content
     content = "[#{dateFormat Date.now(), 'HH:MM:ss'}] #{content}"
+
     @editor.addContent content
+    debug 'log', content
+
+    if content[content.length - 3..] is '...'
```
Comment (usirin) [inline client/new-stack-editor/lib/controllers/logs.coffee]: A comment on what this is doing would be really helpful here.

File: client/app/lib/util/stacks/updatecustomvariable.coffee (hunk 1 above)
Comment (gokmen) [inline client/app/lib/util/stacks/updatecustomvariable.coffee]: right, missed that one. updating.

**Review by sinan — APPROVED** [review]
👍

File: client/app/lib/remote-extensions/stacktemplate.coffee [hunk 3]
```diff
@@ -3,9 +3,19 @@ remote = require('../remote')
 module.exports = class JStackTemplate extends remote.api.JStackTemplate
 
 
-  getCredentialIdentifiers: ->
+  getCredentialIdentifiers: (exclude = ['custom']) ->
 
     ids = []
-    ids = ids.concat id  for provider, id of (@credentials ? {})
+    for provider, id of (@credentials ? {}) when provider not in exclude
+      ids = ids.concat id
```
Comment (gokmen) [inline client/app/lib/remote-extensions/stacktemplate.coffee]: one of them getting the `value` the other one is getting the `key` excluded

**Pull request merged**